### **GITHUB** **CONNECTION**

In [1]:
!git clone https://github.com/Donerne/6100-Project-Two.git

Cloning into '6100-Project-Two'...
remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (6/6), done.
remote: Compressing objects: 100% (4/4), done.
Receiving objects: 100% (6/6), done.
Resolving deltas: 100% (1/1), done.
remote: Total 6 (delta 1), reused 0 (delta 0), pack-reused 0 (from 0)


In [2]:
%cd 6100-Project-Two

/content/6100-Project-Two


### **IMPORT DEPENDENCIES**

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### **DATA PREPROCESSING**

In [6]:
#Mount Google Drive in Colab
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [7]:
# Upload the datasets
from google.colab import files
uploaded = files.upload()


Saving Project2_Test.xlsx to Project2_Test.xlsx
Saving Project2_Training.xlsx to Project2_Training.xlsx


In [8]:
# Load the datasets
train_df = pd.read_excel("Project2_Training.xlsx")
test_df = pd.read_excel("Project2_Test.xlsx")

In [9]:
# Display the first few rows of each dataset
print("Training Data:")
print(train_df.head())
print("\nTest Data:")
print(test_df.head())

Training Data:
   QUOTEDATE  VEHICLEYEAR          VEHICLEMAKE  \
0 2016-01-02         2006           VOLKSWAGEN   
1 2016-01-02         2017              HYUNDAI   
2 2016-01-02         2015  CHEVROLET TRUCK/VAN   
3 2016-01-02         2010               TOYOTA   
4 2016-01-04         2013                HONDA   

                        VEHICLEMODEL  ANNUAL_KM  COMMUTE_DISTANCE  \
0                       JETTA 2.5 4P    10000.0               NaN   
1                     ELANTRA GL 4DR    20000.0               NaN   
2  SILVERADO 2500 HD LT CREW CAB 4WD    15000.0               NaN   
3                     COROLLA CE 4DR    20000.0              20.0   
4                       CIVIC LX 4DR    30000.0              30.0   

  MARKING_SYSTEM TRACKING_SYSTEM VEHICLE_OWNERSHIP  VEHICLE_VALUE  ...  \
0            NaN             NaN             Owned         3000.0  ...   
1            NaN             NaN             Owned            1.0  ...   
2            NaN             NaN               

In [10]:
train_df.info()
test_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 101891 entries, 0 to 101890
Data columns (total 26 columns):
 #   Column                          Non-Null Count   Dtype         
---  ------                          --------------   -----         
 0   QUOTEDATE                       101891 non-null  datetime64[ns]
 1   VEHICLEYEAR                     101891 non-null  int64         
 2   VEHICLEMAKE                     101891 non-null  object        
 3   VEHICLEMODEL                    101891 non-null  object        
 4   ANNUAL_KM                       101890 non-null  float64       
 5   COMMUTE_DISTANCE                66478 non-null   float64       
 6   MARKING_SYSTEM                  25 non-null      object        
 7   TRACKING_SYSTEM                 4 non-null       object        
 8   VEHICLE_OWNERSHIP               25441 non-null   object        
 9   VEHICLE_VALUE                   37218 non-null   float64       
 10  VEHICLEUSE                      101891 non-null  object 

In [11]:
train_df.describe()
test_df.describe()

QUOTEDATE   VEHICLEYEAR       ANNUALKM  \
count                          15114  15114.000000   15114.000000   
mean   2017-01-29 17:00:01.429138432   2009.461559   14834.406577   
min              2017-01-03 00:00:00   1931.000000       0.000000   
25%              2017-01-16 00:00:00   2006.000000   10000.000000   
50%              2017-01-30 00:00:00   2010.000000   15000.000000   
75%              2017-02-13 00:00:00   2014.000000   20000.000000   
max              2017-02-27 00:00:00   2018.000000  500000.000000   
std                              NaN      5.707476   10095.170809   

       COMMUTEDISTANCE   VEHICLEVALUE   YEAROFBIRTH  YEARSLICENSED  \
count      9923.000000    4955.000000  15114.000000   15111.000000   
mean         14.273304   20889.318264   1991.758502      20.639137   
min           0.000000       1.000000   1900.000000       0.000000   
25%           6.000000    5000.000000   1962.000000       6.000000   
50%          10.000000   20000.000000   1976.000000      17.000000   
75%          16.000000   30539.000000   1988.000000      33.000000   
max         600.000000  200000.000000   9999.000000      95.000000   
std          16.821911   17718.220513    374.922672      16.126238   

       YEARSASPRINCIPALDRIVER  CONVICTIONCOUNTMINOR3YRS  \
count              168.000000              15114.000000   
mean                 9.136905                  0.159455   
min                  0.000000                  0.000000   
25%                  4.000000                  0.000000   
50%                 10.000000                  0.000000   
75%                 10.000000                  0.000000   
max                 40.000000                 11.000000   
std                  7.044704                  0.517139   

       CONVICTIONCOUNTMAJOR3YRS  CONVICTIONCOUNTCRIMINAL3YRS  \
count              15114.000000                 15114.000000   
mean                   0.003507                     0.005293   
min                    0.000000                     0.000000   
25%                    0.000000                     0.000000   
50%                    0.000000                     0.000000   
75%                    0.000000                     0.000000   
max                    2.000000                     2.000000   
std                    0.064469                     0.076123   

       ASSIGNEDLOSSESPD5YRS  SUSPENSIONCOUNT  
count          15114.000000     15114.000000  
mean               0.057695         0.007477  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         0.000000  
75%                0.000000         0.000000  
max                2.000000         2.000000  
std                0.241536         0.099011

#### Checking For Duplicates

In [12]:
# Check for duplicates in the training dataset
train_duplicates = train_df.duplicated()
print("Duplicate rows in training data:")
print(train_duplicates.value_counts())

# Check for duplicates in the test dataset
test_duplicates = test_df.duplicated()
print("\nDuplicate rows in test data:")
print(test_duplicates.value_counts())

Duplicate rows in training data:
False    101886
True          5
Name: count, dtype: int64

Duplicate rows in test data:
False    15059
True        55
Name: count, dtype: int64


OUTPUT INTERPRETATION

There are 5 duplicated rows in the training dataframe and 101,886 unique rows. There are 55 duplicated rows in the test dataframe and 15,059 unique rows.

In [21]:
# Configuring user details
!git config --global user.email "kelechiajah1@gmail.com"
!git config --global user.name "stephanieajah"
# Remove existing GitHub remote URL
!git remote remove origin

# Add it back with the token embedded (replace YOUR_TOKEN_HERE with your actual token)
!git remote add origin https://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O@github.com/Donerne/6100-Project-Two.git


In [25]:
# Generate SSH key
!ssh-keygen -t rsa -b 4096 -C "kelechiajah1@gmail.com" -f ~/.ssh/id_rsa -N ""

# Display the public key so you can add it to GitHub
!cat ~/.ssh/id_rsa.pub



Generating public/private rsa key pair.
Created directory '/root/.ssh'.
Your identification has been saved in /root/.ssh/id_rsa
Your public key has been saved in /root/.ssh/id_rsa.pub
The key fingerprint is:
SHA256:r6MKFDMGMbNCn573VD0qn/xHi7AqU9y2csJy3HAWNPc kelechiajah1@gmail.com
The key's randomart image is:
+---[RSA 4096]----+
|=o               |
|.=. .    o..     |
|o =o    ..oo.    |
|...+.   ... .E   |
|  .o ..oS..      |
| .  . o++B.  .   |
|  .   +.*+= o .  |
|   . + *.*.. o   |
|    ..*o*. ..    |
+----[SHA256]-----+
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQDTEGsze06szB5sfiVVnTHuNmO8+HyfDjnGLDxxb7hQfUBjoBgkrCuvH574sEVTqZEGadOXuiaYKzmKoVdXJ9Vm33Qy2KfUVPoIlyhedLcWJGazWmZpyhwod1mKRJrnloi5BhvnqD8kPWZoN3CPx1PEPxPEVYJ9kyRXAej0QrZgOQQomvLEPpmiutFOeMzVCc8nJBi4jnLZiiIjqg0mvljVK8VhbG9Ud7c5TA6XHJ2/bKeqIxhg5GLsEPQN/gVVMZtM2xKeN2uAu3dhqTu45BtBSzat6zCy2/ejK50d/HAc1BjbvdFqIi0TCdI++SQfSmS7hMovkcaKAKpO+X9V01zGNb6hmbGLsGdvhYaVqcfpbDWaPH26iwEU/0B1H5UC1ZL0Zt+q8T3r7gSTi9OfqX87K5tWXbrQ5ol4nI3cHXyQ1hHv

In [26]:
!git remote set-url origin git@github.com:Donerne/6100-Project-Two.git


In [28]:
!rm -rf ~/.ssh/known_hosts
!ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts


# github.com:22 SSH-2.0-babeld-44e29decf


In [34]:
import os
os.listdir('/content/drive/My Drive/Colab Notebooks/')


['Project 1 DATA*6100.ipynb',
 'Copy of Project 1 DATA*6100.ipynb',
 'PythonAIAssignment.ipynb',
 'nonogram_test.cpp',
 'nonogram.h.ipynb',
 'calculator.py',
 'Untitled (1)',
 'House Price Prediction Project.ipynb',
 'AI FASHION PROJECT.ipynb',
 'makenonogram.ipynb',
 'nonogram.ipynb',
 'Untitled',
 'Insurance_Purchase_Prediction_Model.ipynb']

In [36]:
!ls '/content/drive/My Drive/Colab Notebooks/'

'AI FASHION PROJECT.ipynb'		     makenonogram.ipynb		  PythonAIAssignment.ipynb
 calculator.py				     nonogram.h.ipynb		  Untitled
'Copy of Project 1 DATA*6100.ipynb'	     nonogram.ipynb		 'Untitled (1)'
'House Price Prediction Project.ipynb'	     nonogram_test.cpp
 Insurance_Purchase_Prediction_Model.ipynb  'Project 1 DATA*6100.ipynb'


In [29]:
!git add .

# Commit the changes
!git commit -m "Loaded datasets and initial code setup. I also checked the data structure and how many duplicates we have."


# Push to GitHub
!git push origin main

On branch main
nothing to commit, working tree clean
Enumerating objects: 5, done.
Counting objects: 100% (5/5), done.
Delta compression using up to 2 threads
Compressing objects: 100% (4/4), done.
Writing objects: 100% (4/4), 11.36 MiB | 9.41 MiB/s, done.
Total 4 (delta 0), reused 0 (delta 0), pack-reused 0
To github.com:Donerne/6100-Project-Two.git
   fb60b62..fba696d  main -> main
